package/MDAnalysis/coordinates/DCD.py

```diff
@@ -89,438 +69,214 @@
-        # seems to do the job on Mac OS X 10.6.4 ... but I have no idea why,
-        # given that the C code seems to define them as normal integers
-        desc = [
-            'file_desc', 'header_size', 'natoms', 'nsets', 'setsread', 'istart',
-            'nsavc', 'delta', 'nfixed', 'freeind_ptr', 'fixedcoords_ptr',
-            'reverse', 'charmm', 'first', 'with_unitcell']
-        return dict(zip(desc, struct.unpack("LLiiiiidiPPiiii", self._dcd_C_str)))
-
-    def write_next_timestep(self, ts=None):
-        """Write a new timestep to the DCD file.
-
         Parameters
         ----------
-        ts : `Timestep` (optional)
-            `Timestep` object containing coordinates to be written to DCD file;
-            by default it uses the current `Timestep` associated with the
-            Writer.
-
-        Raises
-        ------
-        :exc:`ValueError`
-           if wrong number of atoms supplied
-        :exc:`~MDAnalysis.NoDataError`
-           if no coordinates to be written.
-
+        filename : str
+            trajectory filename
+        convert_units : bool (optional)
+            convert units to MDAnalysis units
+        dt : float (optional)
+            overwrite time delta stored in DCD
+        **kwargs : dict
+            General reader arguments.
 
-        .. versionchanged:: 0.7.5
-           Raises :exc:`ValueError` instead of generic :exc:`Exception`
-           if wrong number of atoms supplied and :exc:`~MDAnalysis.NoDataError`
-           if no coordinates to be written.
 
+        .. versionchanged:: 0.17.0
+           Changed to use libdcd.pyx library and removed the correl function
         """
-        if ts is None:
-            try:
-                ts = self.ts
-            except AttributeError:
-                raise NoDataError("DCDWriter: no coordinate data to write to trajectory file")
-        if not ts.n_atoms == self.n_atoms:
-            raise ValueError("DCDWriter: Timestep does not have the correct number of atoms")
-        unitcell = self.convert_dimensions_to_unitcell(ts).astype(np.float32)  # must be float32 (!)
-        if not ts._pos.flags.f_contiguous:  # Not in fortran format
-            ts = Timestep.from_timestep(ts)  # wrap in a new fortran formatted Timestep
+        super(DCDReader, self).__init__(
+            filename, convert_units=convert_units, **kwargs)
+        self._file = DCDFile(self.filename)
+        self.n_atoms = self._file.header['natoms']
+
+        delta = mdaunits.convert(self._file.header['delta'],
+                                 self.units['time'], 'ps')
+        if dt is None:
+            dt = delta * self._file.header['nsavc']
+        self.skip_timestep = self._file.header['nsavc']
+
+        self._ts_kwargs['dt'] = dt
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        frame = self._file.read()
+        # reset trajectory
+        if self._file.n_frames > 1:
+            self._file.seek(1)
```

Comment:
This is an oddity of MDAnalysis. We always load the first frame. So when we open any trajectory I have to say the seek functions have already read one frame and the next frame to read is frame 1. Remember seek always puts us so the next read call returns the frame we seemed to.